pkg/descheduler/strategies/topologyspreadconstraint_test.go

```diff
@@ -0,0 +1,103 @@
+package strategies
+
+import (
+	"context"
+	"sigs.k8s.io/descheduler/pkg/api"
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	core "k8s.io/client-go/testing"
+	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
+	"sigs.k8s.io/descheduler/test"
+)
+
+func TestTopologySpreadConstraint(t *testing.T) {
```

Comment:
We need more cases by mixing the following conditions:

- multiple constraints
- qualified topology domains that have 0 matching pods
- pod with topologySpreadConstraint matches or not match itself on its labelSelector
- pod has nodeSelector/nodeAffinity